lib/mix/test/mix/tasks/deps_test.exs

```diff
@@ -251,6 +251,18 @@ defmodule Mix.Tasks.DepsTest do
     end
   end
 
+  test "fails with message on missing dependencies" do
+      message = "\"mix deps.unlock\" expects dependencies as arguments or " <>
+                "a flag indicating which dependencies to unlock. " <>
+                "The --all option will unlock all dependencies while " <>
+                "the --unused option unlocks unused dependencies"
+
+      assert_raise Mix.Error, message, fn ->
+        Mix.Tasks.Deps.Unlock.run []
+      end
```

Comment:
Maybe we can only test part of this message, just to test that it's raised, instead of testing all of it? Something like

```elixir
assert_raise Mix.Error, ~r/"mix deps\.unlock" expects dependencies as arguments/, fn ->
  Mix.Tasks.Deps.Unlock.run []
end
```

Wdyt?